tests: bluetooth: shell: Revert back to default logging settings

carlescufi opened the pull request:
Disable the CONFIG_TEST_LOGGING_DEFAULTS Kconfig option which forces
immedate logging, since that is not compatible with Bluetooth and its
shell.

carlescufi committed:
tests: bluetooth: shell: Revert back to default logging settings

Disable the CONFIG_TEST_LOGGING_DEFAULTS Kconfig option which forces
immedate logging, since that is not compatible with Bluetooth and its
tests.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

Vudentz commented:
So the fixes for the LOG_IMMEDIATE didn't work?

carlescufi commented:
> So the fixes for the LOG_IMMEDIATE didn't work?

Well `LOG_IMMEDIATE` crashes the Link Layer due to disabling interrupts for too long during printing (`put_sync_hexdump`). So it works but it creates too much IRQ latency for the controller to handle.

joerchan commented:
> So the fixes for the LOG_IMMEDIATE didn't work?

@Vudentz The controller cannot handle the ISR latency added by LOG_IMMEDIATE.
It will give you 10 ms ISR latency and the controller will crash.

jhedberg reviewed (changes requested):
I suppose `tests/bluetooth/mesh_shell` needs fixing as well?

Vudentz commented:
> > So the fixes for the LOG_IMMEDIATE didn't work?
> 
> @Vudentz The controller cannot handle the ISR latency added by LOG_IMMEDIATE.
> It will give you 10 ms ISR latency and the controller will crash.

This should depend on the controller being enable then, does LOG_MINIMAL would work then? It kind bad not to be able to get all the traces for combined builds though.

carlescufi committed:
Bluetooth: common: Statically check for immediate logging

Immediate logging is not compatible with the software-based controller
due to the additional ISR latency that it introduces. Ensure that
deferred logging is in use whenever using the software-based LL.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
> I suppose `tests/bluetooth/mesh_shell` needs fixing as well?

Changed the approach, please take another look

carlescufi commented:
> > > So the fixes for the LOG_IMMEDIATE didn't work?
> > 
> > 
> > @Vudentz The controller cannot handle the ISR latency added by LOG_IMMEDIATE.
> > It will give you 10 ms ISR latency and the controller will crash.
> 
> This should depend on the controller being enable then, does LOG_MINIMAL would work then? It kind bad not to be able to get all the traces for combined builds though.

Yes, that is what we've done now, with a conditional build assert based on the controller being enabled.

carlescufi commented:
@jhedberg @Vudentz I have simplified this. Once `CONFIG_LOG_MINIMAL` comes in, that will be an acceptable mechanism for logging, but not `CONFIG_LOG_IMMEDIATE`.

Vudentz reviewed (approved):
LGTM

aescolar commented on subsys/bluetooth/common/dummy.c:
```diff
@@ -28,3 +28,13 @@ BUILD_ASSERT(CONFIG_BT_HCI_TX_PRIO < CONFIG_BT_RX_PRIO);
+ */
+#if !defined(CONFIG_TEST) && (defined(CONFIG_BT_LL_SW_SPLIT) || \
+			      defined(CONFIG_BT_LL_SW_LEGACY))
+BUILD_ASSERT_MSG(!IS_ENABLED(CONFIG_LOG_IMMEDIATE), "Immediate logging not \
```
This will hit in nrf52_bsim builds, as there it is the default, If you still want this check, can you maybe filter out by POSIX_ARCH?

carlescufi replied:
I see, thanks. I will fix it here instead, I'd rather run with immediate logging in native posix.

carlescufi replied:
Fixed

aescolar commented:
Ok, it has already crashed: https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/52154/1/console

aescolar commented on subsys/bluetooth/common/dummy.c:
```diff
@@ -28,3 +28,13 @@ BUILD_ASSERT(CONFIG_BT_HCI_TX_PRIO < CONFIG_BT_RX_PRIO);
+#if !defined(CONFIG_TEST) && (defined(CONFIG_BT_LL_SW_SPLIT) || \
+			      defined(CONFIG_BT_LL_SW_LEGACY))
+BUILD_ASSERT_MSG(!IS_ENABLED(CONFIG_LOG_IMMEDIATE), "Immediate logging not \
+		 supported with the software Link Layer");
```
Silly thing, these tabs are also printed, so maybe chop the string into 2 concatenated ones, i.e.
"Immediate logging not "
"supported with the software Link Layer"

carlescufi replied:
Changed that now

andrewboie commented:
> Once CONFIG_LOG_MINIMAL comes in, that will be an acceptable mechanism for logging, but not CONFIG_LOG_IMMEDIATE.

This landed, is this PR still applicable?

carlescufi commented:
> > Once CONFIG_LOG_MINIMAL comes in, that will be an acceptable mechanism for logging, but not CONFIG_LOG_IMMEDIATE.
> 
> This landed, is this PR still applicable?

It is, because `CONFIG_LOG_MINIMAL` is still not suitable for out test shell.

andrewboie commented:
@carlescufi right, reason I ask is that CONFIG_TEST_LOGGING_DEFAULTS now selects minimal logging instead of immediate, so is it still necessary to disable CONFIG_TEST_LOGGING_DEFAULTS for these tests?

carlescufi commented:
> @carlescufi right, reason I ask is that CONFIG_TEST_LOGGING_DEFAULTS now selects minimal logging instead of immediate, so is it still necessary to disable CONFIG_TEST_LOGGING_DEFAULTS for these tests?

Thanks for checking back. It is because those "tests" are used regularly by developers to test newly developed features. They are more like a toolbox, and `CONFIG_LOG_MINIMAL` is definitely not feature-rich enough to handle that. In addition I assume that `printk()` disables interrupts when printing just like `LOG_IMMEDIATE` does, and that will make the controller crash.
